### DISCUSSION TITLE
JsonArray filtering

### DISCUSSION THREAD

**Discussion by skittlesaur** [prisma/prisma#15711] [Q&A]
I have a `targets` property in my `schema.prisma` that is `Json[]`. The json object contains a property `id` which i want to query the elements that contain an id of an element.

For example, if i have an input of id `123`, i want to get all records that has the id `123` in some element in the json array of targets.

What i tried so far:

```js
const { id } = args

const activity = await ctx.prisma.activity.findMany({
    where: {
        {
          targets: {
            has: {
              path: ['id'],
              equals: id,
            }
          }
        }
    },
  })
```
This always returns an empty array.

**Comment by austincrim** [comment]
Hey @skittlesaur 👋,

Could you write out a sample record that includes the `targets` field and what result you want to get? I will try to recreate it 👍.

**Reply by skittlesaur** [reply]
Hey @austincrim, 

Removed unnecessary data for simplification. 

The `schema.prisma` looks something like this:

```
model Activity {
  id          String  @id // ..
  targets Json[]
}
```

A record would look like
```js
{
  id: activity_1,
  targets: [
    {
      id: 123,
      name: "Some name"
    }, 
    {
      id: 531,
      name: "other"
    }
  ]
}
```
```js
{
  id: activity_2
  targets: [
    {
      id: 123,
      name: "Some name"
     }
  ]
}
```
```js
{
  id: activity_3
  targets: [
    {
      id: 4123,
      name: "Blabla"
     }
  ]
}
```

What i wanna filter is any activity that contains a target with the id `123`, in this case, `activity_1` and `activity_2`.

**Reply by austincrim** [reply]
I think the issue here is that the `path` / `equals` API is only available for the `Json` type, not the `Json[]` type. Is there a specific reason you need a scalar list of `Json` fields? If not, I would recommend switching to a plain `Json` type.

**Reply by skittlesaur** [reply]
The thing is, `targets` may contain multiple objects so i need is as a list. I have changed it to `Json` type and verified that it's a list via code, but i still can't figure out how to search the id in the targets array (which is json now). Is that possible anyhow?

**Reply by nurul3101** [reply]
Hey @skittlesaur 👋 
Are you using PostgreSQL or MySQL? Because from the [docs](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#path), Filtering by the key values of objects inside an array is only supported by the MySQL connector.

**Reply by vladi-stevanovic** [reply]
👋 Hi @skittlesaur  , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by denfield-daveyonmayne** [reply]
Any support for MongoDB? Using  [equals ](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-exact-field-value)is not always ideal.

**Reply by nurul3101** [reply]
@denfield-daveyonmayne We have a feature request for it
- https://github.com/prisma/prisma/issues/10879

Adding a 👍 to it would help our team in prioritising it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
